### PR TITLE
Follow up the migration of the replication system's worker

### PR DESCRIPTION
**Pull request by iagaponenko**

### CODE REVIEW

**Comment by fjammes** [comment]
@iagaponenko, thanks so much for the PR! It seems the CI has crashed because a bug has been introduced while changing branch name from "master" to "main"... Could you please double check that the related replication service code is embedded inside this qserv image: "qserv/lite-qserv:2021.10.1-rc1-37-g45646a1cb" (or update https://github.com/lsst/qserv-operator/blob/main/manifests/base/image.yaml) while I'm trying to fix the CI in the current branch?

**Comment by iagaponenko** [comment]
@fjammes I have tested Qserv tag, ad it does have the replication image:
```
docker run -it --rm qserv/lite-qserv:2021.10.1-rc1-37-g45646a1cb bash -c 'ls -al /usr/local/bin/'
total 18592
drwxr-xr-x 1 root  root     4096 Nov  6 04:55 .
drwxr-xr-x 1 root  root       17 Nov  6 04:55 ..
-rwxr-xr-x 1 root  root  2638032 Nov  3 16:29 cmsd
-r-x---r-x 1 qserv qserv    1243 Nov  6 04:30 entrypoint
-rwxr-xr-x 1 root  root      218 Nov  3 16:33 f2py
-rwxr-xr-x 1 root  root      218 Nov  3 16:33 f2py3
-rwxr-xr-x 1 root  root      218 Nov  3 16:33 f2py3.8
-rwxr-xr-x 1 root  root    67176 Nov  3 16:27 mysql-proxy
-rwxr-xr-x 1 root  root      246 Nov  3 16:33 normalizer
-rwxr-xr-x 1 qserv qserv  399872 Nov  6 04:53 qserv-replica-calc-cs
-rwxr-xr-x 1 qserv qserv  402128 Nov  6 04:53 qserv-replica-config
-rwxr-xr-x 1 qserv qserv  402320 Nov  6 04:53 qserv-replica-config-test
-rwxr-xr-x 1 qserv qserv  402328 Nov  6 04:53 qserv-replica-controller-cmd
-rwxr-xr-x 1 qserv qserv  889120 Nov  6 04:53 qserv-replica-file
-rwxr-xr-x 1 qserv qserv 1788184 Nov  6 04:53 qserv-replica-job
-rwxr-xr-x 1 qserv qserv  416416 Nov  6 04:53 qserv-replica-master-http
-rwxr-xr-x 1 qserv qserv 1164296 Nov  6 04:52 qserv-replica-test
-rwxr-xr-x 1 qserv qserv  404704 Nov  6 04:53 qserv-replica-worker
-rwxr-xr-x 1 qserv qserv  401616 Nov  6 04:53 qserv-replica-worker-notify
-rw-r--r-- 1 qserv qserv    3024 Nov  6 04:30 qserv-smig.py
-rwxr-xr-x 1 qserv qserv 1333648 Nov  6 04:37 sph-duplicate
-rwxr-xr-x 1 qserv qserv  640288 Nov  6 04:37 sph-duplicate2
-rwxr-xr-x 1 qserv qserv  602800 Nov  6 04:37 sph-estimate-stats
-rwxr-xr-x 1 qserv qserv  850056 Nov  6 04:37 sph-htm-index
-rwxr-xr-x 1 qserv qserv  557920 Nov  6 04:37 sph-layout
-rwxr-xr-x 1 qserv qserv  880648 Nov  6 04:37 sph-partition
-rwxr-xr-x 1 qserv qserv  850248 Nov  6 04:37 sph-partition-matches
-rwxr-xr-x 1 root  root  3268960 Nov  3 16:29 xrdfs
-rwxr-xr-x 1 root  root   591296 Nov  3 16:29 xrootd
```
Perhaps a problem is that I haven't merged the C++ code of the `qserv-replica-worker` into the `main` branch of Qserv? Note that mny branch https://jira.lsstcorp.org/browse/DM-32810 has two PRs. One is for `qserv-operator`, and the other one is for `qserv`: https://github.com/lsst/qserv/pull/666

You may try this tag for Qserv: `qserv/lite-qserv:2021.10.1-rc1-84-g47af29655`. This tag was made by the CI on the Qserv PR.

**Comment by fjammes** [comment]
@iagaponenko , I have fixed the CI bug, let's see if the CI pass before going further: https://github.com/lsst/qserv-operator/runs/4464051786?check_suite_focus=true (I'll have a look tomorrow)

**Comment by fjammes** [comment]
@iagaponenko looking at the CI logs, it seems the replication worker does not start no more... In fact the Qserv operator use a fixed Qserv version issued from main branch. Maybe you can try `qserv/lite-qserv:2021.10.1-rc1-84-g47af29655` (edit https://github.com/lsst/qserv-operator/blob/main/manifests/base/image.yaml and push your commit) to see what happen? But in the long term I would prefer the CI to use images coming from main Qserv branch (Your Qserv code should then be validated by the Qserv CI, with a fixed version of the qserv-operator). However, this wish might be difficult to grant while developing concurrently on Qserv and qserv-operator.

**Comment by iagaponenko** [comment]
@fjammes sorry for the delay! I have just merged the latest code of `qserv` PR https://github.com/lsst/qserv/pull/666 into the `main` branch. I hope this will resolve the issue.

**Comment by fjammes** [comment]
@iagaponenko , it seems that this fix require some update in qserv-ingest, because the replication service version has been incremented, here is what CI integration tests show (see https://github.com/lsst/qserv-operator/runs/4468579024?check_suite_focus=true for details):
```
ValueError: Invalid replication server version (is 8, expected 7)
```
Could you please create a branch `tickets/DM-32810` in qserv-ingest and implement the upgrade of the replication service client? Or do you want me to work on this? I propose to merge this PR when all Qserv components are up to date and ready to work together on our pre-production platforms, so that we can ship a release if needed.

**Comment by iagaponenko** [comment]
@fjammes actually, the API changed in this branch: https://jira.lsstcorp.org/browse/DM-32395
I have made this trivial PR linked to that branch: https://github.com/lsst-dm/qserv-ingest/pull/19
Could you, please, review the one? You may also merge it to the `main` branch since this was a trivial fix. Or, I can do this.
I'm sorry for missing to upgrade this package!

**Comment by fjammes** [comment]
Hi @iagaponenko , I have validated locally all your updates against qserv-operator, qserv-ingest and qserv. However I tried to rebase your branch against latest qserv-operator main branch and it fails because path to dashboard web files has been changed in latest qserv (ping @fritzm @n8pease ). I'll fix that and let you know ASAP. It is quite painful on my side to integrate qserv, qserv-operator and qserv-ingest when all of them change concurrently...

**Comment by fjammes** [comment]
@iagaponenko Good news! Integration tests are passing. So my review is done. Thanks so much. Integration is quite painful when all components move concurrently, but I do not see how to ease that for now.

File: configmap/repl-wrk/start/start.sh [hunk 1]
```diff
@@ -69,7 +69,7 @@ QSERV_WORKER_DB_URL="mysql://${QSERV_WORKER_DB_USER}:${QSERV_WORKER_DB_PASSWORD}
 entrypoint --log-level DEBUG smig-update --worker-connection "mysql://${QSERV_WORKER_DB_USER}:${QSERV_WORKER_DB_PASSWORD}@${QSERV_WORKER_DB_DN}:${QSERV_WORKER_DB_PORT}"
 
 
-qserv-replica-worker ${WORKER_ID} --config=${CONFIG} --qserv-worker-db="${QSERV_WORKER_DB_URL}" --debug
```
Comment (fjammes) [inline configmap/repl-wrk/start/start.sh:72]: @iagaponenko could you please remove $WORKER_ID definition on top of the script and then merge?

**Review by fjammes — APPROVED** [review]

**Pull request merged**